src/CoreWCF.Primitives/src/CoreWCF/Security/SecurityHeader.cs

```diff
@@ -0,0 +1,157 @@
+using System;
+using System.Globalization;
+using CoreWCF.Channels;
+using CoreWCF.Description;
+
+namespace CoreWCF.Security
+{
+    internal abstract class SecurityHeader : MessageHeader
+    {
+        private readonly string actor;
+        private readonly SecurityAlgorithmSuite algorithmSuite;
+        private bool encryptedKeyContainsReferenceList = true;
+        private Message message;
+        private readonly bool mustUnderstand;
+        private readonly bool relay;
+        private bool requireMessageProtection = true;
+        private bool processingStarted;
+        private bool maintainSignatureConfirmationState;
+        private readonly SecurityStandardsManager standardsManager;
+        private MessageDirection transferDirection;
+        private SecurityHeaderLayout layout = SecurityHeaderLayout.Strict;
+
+        public SecurityHeader(Message message,
+            string actor, bool mustUnderstand, bool relay,
+            SecurityStandardsManager standardsManager
+            , SecurityAlgorithmSuite algorithmSuite,
+            MessageDirection transferDirection)
+        {
+            if (message == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("message");
```

Comment:
nameof. There is also an assignment pattern that can be used here that VS has a code refactor helper for. This can be converted to:
```c#
this.message = message ?? throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(message));
```
When message is null, it executes the code after the ?? to retrieve a default value, but it actually throws instead.